### DISCUSSION TITLE
Date docs indicates some implicit behavior

### DISCUSSION THREAD

**Discussion by O1O1O1O** [ClickHouse/ClickHouse#41392] [General]
The [documentation](https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/data-types/date.md) for `Date` is misleading as it says dates use a two byte "days since 1970-01-01" epoch value.  However the example for inserting a date using an integer value (vs. a string) is clearly not using a 2 byte value for the days as it inserts `1546300800` and gets back the date `2019-01-01`.  The input number is > the 2 byte integer range, and 2019-01-01 is 17897 days from the Unix Epoch:

```
select toInt32(toDate('2019-01-01'));
17897
```

Initially I thought because `1546300800` ends with `0800` it has a timezone offset in it but that also doesn't work.  Then I realized 2019-01-01T00:00:00Z is also 1546300800 in seconds since the epoch.   So the example appears to be using an Int32 timestamp.  

But then I discovered if you use the days since epoch that works too:
```
select toDate(17897)
2019-01-01
```

There are 86400 seconds in a day so if I do 
```
select toDate(86400)
2019-01-02
```

and the max unsigned 2-byte value is 65535
```
select toDate(65535)
2149-06-06
```

if I add one more then:
```
select toDate(65536)
1970-01-01
```

So it seems that if the numeric value exceeds the range of UInt16 then it is interpreted as a seconds since epoch timestamp, otherwise it is interpreted as the days since epoch as documented.  If I convert a date back to an UInt16 or Int32 I aways get the days since epoch value:

```
select toUInt16(toDate('1970-01-02'))
select toInt32(toDate('1970-01-02'))
```
both give 1.

Based on the above I've no doubt it is stored as an unsigned int internally.   However there seems to be some implicit behavior when converting a numeric value to a date and the example in the docs should probably show and explain both behaviors.  That's assuming this isn't a bug?

**Comment by UnamedRus** [comment]
> That's assuming this isn't a bug?

It's not a bug, you are right. 

Docs should include examples of this behavior.
